apps/explorer/src/utils/getName.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO : Import from  SDK
+import { VALDIATOR_NAME } from '~/pages/validator/ValidatorDataTypes';
+
+export function getName(rawName: string | number[]) {
```

Comment:
I am not sure about this one. Seems to be used in the specific case of the validator name, where it should be string already from the API right? Maybe it's a bug there?

Also is there a case where it is a base64 string?

Reply:
I know this is marked as resolved but replying here for the sake of completeness: There are actually 3 possible return values based on the version of Sui we talk to: 
1. Base64 encoded string. This was the original API response for `vector<u8>`.
2. An array of u8 bytes. This replaced the previous base64 encoding in Sui.
3. An actual string, which is dependent on validator metadata moving from `vector<u8>` to `string`, but should happen at some point.

We can probably remove Base64 encoding right now because I think we only talk to networks that have that change rolled out, and at some point we can remove this logic entirely.

Reply:
Since we test for `/^[A-Z-_.\s0-9]+$/i.test` string and return the rawName in those instance, I think we are covered. Although I am thinking of switching  `buffer.from` to `decodeURIComponent(atob(raw))` or `textDecoder.decode(new Base64DataBuffer(rawName).getData())`  since buffer isn't supported on the browser